### PR TITLE
MAM-4071-interactions-going-up-but-not-back-down

### PR DESCRIPTION
**Pull request by derspyy**
sorry for the previous oversight 😅

### CODE REVIEW

File: Mammoth/Models/PostCardModel.swift [hunk 1]
```diff
@@ -888,22 +888,22 @@ extension PostCardModel {
...
     
     public func unlikeTap() -> Void {
-        guard case .mastodon(let status) = data, status.favourited == true else { return }
+        guard case .mastodon(let status) = data, (status.reblog?.favourited ?? status.favourited) == true else { return }
         (status.reblog ?? status).unlikeTap()
     }
     
     public func repostTap() -> Void {
-        guard case .mastodon(let status) = data, status.reblogged == false else { return }
+        guard case .mastodon(let status) = data, (status.reblog?.favourited ?? status.favourited) == false else { return }
```
Comment (bnolens) [inline Mammoth/Models/PostCardModel.swift:901]: This line is supposed to be `guard case .mastodon(let status) = data, (status.reblog?.reblogged ?? status.reblogged ?? false) == false else { return }`

File: Mammoth/Models/PostCardModel.swift [hunk 2]
```diff
@@ -888,22 +888,22 @@ extension PostCardModel {
...
         (status.reblog ?? status).repostTap()
     }
     
     public func unrepostTap() -> Void {
-        guard case .mastodon(let status) = data, status.reblogged == true else { return }
+        guard case .mastodon(let status) = data, (status.reblog?.favourited ?? status.favourited) == true else { return }
```
Comment (bnolens) [inline Mammoth/Models/PostCardModel.swift:906]: This line is supposed to be `guard case .mastodon(let status) = data, (status.reblog?.reblogged ?? status.reblogged ?? false) == true else { return }`

**Review by bnolens — CHANGES_REQUESTED** [review]
Great catch @derspyy. Please review my comment for an additional fix.

**Comment by bnolens** [comment]
I pushed the additional changes in https://github.com/TheBLVD/mammoth/pull/529/commits/4ee1f36ed8804a905dbaa6e863e83d74edcb77be

**Review by bnolens — APPROVED** [review]
LGTM

File: Mammoth/Models/PostCardModel.swift (hunk 2 above)
Comment (derspyy) [reply Mammoth/Models/PostCardModel.swift]: can you plz make these changes? can't rn...

Comment (bnolens) [reply Mammoth/Models/PostCardModel.swift]: Already did

**Pull request merged**